apps/backend/src/modules/users/user.repository.ts

```diff
@@ -1,27 +1,53 @@
+import { transaction } from "objection";
+
+import { RelationName } from "~/libs/enums/relation-name.enum.js";
 import { type Repository } from "~/libs/types/types.js";
 import { UserEntity } from "~/modules/users/user.entity.js";
 import { type UserModel } from "~/modules/users/user.model.js";
 
+import { UserDetailsModel } from "./user-details.model.js";
+
 class UserRepository implements Repository {
+	private userDetailsModel: typeof UserDetailsModel;
+
 	private userModel: typeof UserModel;
-	public constructor(userModel: typeof UserModel) {
+
+	public constructor(
+		userModel: typeof UserModel,
+		userDetailsModel: typeof UserDetailsModel,
+	) {
 		this.userModel = userModel;
+		this.userDetailsModel = userDetailsModel;
 	}
 
 	public async create(entity: UserEntity): Promise<UserEntity> {
-		const { email, passwordHash, passwordSalt } = entity.toNewObject();
+		const { email, name, passwordHash, passwordSalt } = entity.toNewObject();
 
-		const user = await this.userModel
-			.query()
-			.insert({
-				email,
-				passwordHash,
-				passwordSalt,
-			})
-			.returning("*")
-			.execute();
+		const userWithDetails = await transaction(
+			this.userModel.knex(),
+			async (trx) => {
+				const user = await this.userModel
+					.query(trx)
+					.insert({
+						email,
+						passwordHash,
+						passwordSalt,
+					})
+					.returning("*");
+
+				const details = await this.userDetailsModel
+					.query(trx)
+					.insert({
+						name,
+						userId: user.id,
+					})
+					.returning("*");
 
-		return UserEntity.initialize(user);
+				return { ...user, ...details };
+			},
+		);
+
+		return UserEntity.initialize(userWithDetails);
```

Comment:
```suggestion
		return UserEntity.initialize({
		  id: user.id,
		  createdAt: user.createdAt,
		  ...
		});
```

Each field should be added separately

Reply:
UserEntity currently doesn't have createdAt logic as in the suggestion above. Should I add logic for createdAt and updatedAdd to the UserEntity?

Updated code with separate arguments looks like that for now:

```
return UserEntity.initialize({
			email: user.email,
			id: user.id,
			name: details.name,
			passwordHash: user.passwordHash,
			passwordSalt: user.passwordSalt,
		})
```

Reply:
Yes, you should add logic for createdAt and updatedAt. You should add them to the updated code, that you shared